nixos/tests/common/systemd-journal-remote/ca.cert.pem

```diff
@@ -0,0 +1,20 @@
+-----BEGIN CERTIFICATE-----
```

Comment:
We don't really care about the specific certificate we use, do we? So instead we should generate them with runCommand to also make sure they never expire.

Reply:
hmm, I'm not sure if this is a good idea, since the generation uses randomness, CI / people will still get expired certificates if they don't garbage collect

Reply:
Which makes me think that the generate cert should really be a `nix-shell` script instead, I'll change it to that.

Reply:
@minijackson The certificates are now committed to the repo, and will still expire - and we'll suddenly have these VM tests fail once that day comes :-/

I think the certificates should be created at VM bootup, and then be distributed to the different machines.

Can you run change the machine configs to point to hard-coded paths outside the nix store, and actually do the cert generation through the test driver?

cc @m1cr0man, as this might be something to do in the ACME tests as well.

Reply:
I think the expiry date on these is waaay in the future. Iirc the main reason to commit these to git is that you want to be able to add the same set of trusted CA certs to each test VM on creation. However, I'm in full agreement that we should avoid this. We can probably make it so that the CA certs can be generated + distributed at build time. I'll look into this for the acme tests and report back

Reply:
Heh and I was quickly reminded why it was done this way: https://github.com/NixOS/nixpkgs/pull/91121#discussion_r505410235 

So I'm indifferent about keeping or not keeping the certs in the store. On one hand, it avoids the issue described in that linked comment chain, on the other it's very unlikely that the tests would run across hosts in such a way that two different sources of that derivation may be used in a single execution? The test suite and VM images are all built at once, so despite the possibility of the derivation being impure surely it'll never cause an issue in practise?